### PR TITLE
Support for Scala Native

### PR DESCRIPTION
**Pull request by alonsodomin**
This is an initial stab at adding support for Scala Native. Some observations on this:

 * Scala Native does not support 2.12 yet, so the 2.11 version is forced
 * Scala Native does not support weak hashmaps, so the `PositionCache` has been implemented the same way as it was in ScalaJS (no cache).

### CODE REVIEW

**Comment by xuwei-k** [comment]
Any updates?

**Comment by alonsodomin** [comment]
The main stopper here is the fact that the build in Travis is required to pass in OpenJDK 6. Scala Native requires JDK 7+ since the compiler plugin uses the NIO file system API. So if the maintainers do not want to drop JDK 6 support, then this PR goes nowhere.

Besides, there is still some work pending even dropping support for JDK 6. Mainly run the scala native build only on Scala 2.11.x and avoid the tests. I would like to have some feedback from the maintainers though on whether this is a piece of work they would be interested on having or not.

File: build.sbt [hunk 1]
```diff
@@ -43,6 +44,10 @@ lazy val `scala-parser-combinators` = crossProject.in(file(".")).
   jsSettings(
     name := "scala-parser-combinators-js"
   ).
+  nativeSettings(
+    name := "scala-parser-combinators-native",
+    scalaVersion := "2.11.11"
```
Comment (gourlaysama) [inline build.sbt]: Something like this should fix the whole Travis issue:

```diff
-    scalaVersion := "2.11.11"
+   skip in compile := !scalaVersion.value.startsWith("2.11") || System.getProperty("java.version").startsWith("1.6"),
+   test := {}
```

**Comment by lrytz** [comment]
@alonsodomin @gourlaysama if you can think of anything in https://github.com/scala/sbt-scala-module that would make this easier and apply to other modules, there's no problem in releasing a new version there.

**Comment by alonsodomin** [comment]
@gourlaysama @lrytz got it passing in the end, had to remove all `org.scala-native` artifacts when the Scala version was not 2.11 since, for some reason, was trying to download the compiler plugin.

A bit hacky but it seems to work. Maybe worth adding these settings to `sbt-scala-module` using a `scalaModuleSettingsNative` so they can be reused...:

```
scalaVersion := "2.11.11",
skip in compile := System.getProperty("java.version").startsWith("1.6"),
test := {},
libraryDependencies := {
  if (!scalaVersion.value.startsWith("2.11"))
    libraryDependencies.value.filterNot(_.organization == "org.scala-native")
  else libraryDependencies.value
}
```

**Comment by gourlaysama** [comment]
@alonsodomin could you rebase, #120 introduced some conflicts. I'll merge this and release 1.0.7 soon after that :)

**Pull request merged**